inst/include/model_simulation.hpp

```diff
@@ -236,96 +210,76 @@ void run_art_progression_and_mortality(int hiv_step,
-  const auto hIDX_15PLUS = pars.base.options.hIDX_15PLUS;
+  constexpr auto ss_b = StateSpace<ModelVariant>().base;
+  const auto& p_nh = pars.base.natural_history;
+  const auto& p_tx = pars.base.art;
+  const auto& p_op = pars.base.options;
+  auto& n_ba = state_next.base;
+  auto& i_ba = intermediate.base;
 
-  for (int g = 0; g < ss.NS; ++g) {
+  for (int g = 0; g < ss_b.NS; ++g) {
+    i_ba.Xart_15plus = 0.0;
 
-    intermediate.base.Xart_15plus = 0.0;
+    i_ba.artelig_hm.setZero();
+    i_ba.Xartelig_15plus = 0.0;
 
-    intermediate.base.artelig_hm.setZero();
-    intermediate.base.Xartelig_15plus = 0.0;
+    i_ba.expect_mort_artelig_hm.setZero();
+    i_ba.expect_mort_artelig15plus = 0.0;
 
-    intermediate.base.expect_mort_artelig_hm.setZero();
-    intermediate.base.expect_mort_artelig15plus = 0.0;
-    
-    for (int ha = hIDX_15PLUS; ha < ss.hAG; ++ha) {
-      for (int hm = intermediate.base.everARTelig_idx; hm < ss.hDS; ++hm) {
-	
-        if (hm >= intermediate.base.anyelig_idx) {
+    for (int ha = p_op.hIDX_15PLUS; ha < ss_b.hAG; ++ha) {
+      for (int hm = i_ba.everARTelig_idx; hm < ss_b.hDS; ++hm) {
+        if (hm >= i_ba.anyelig_idx) {
           // TODO: Implement special population ART eligibility
           real_type prop_elig = 1.0;
-
-	  real_type tmp_artelig = prop_elig * state_next.base.h_hiv_adult(hm, ha, g);
-          intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS) = tmp_artelig;
-	  intermediate.base.artelig_hm(hm) += tmp_artelig;
-	  intermediate.base.Xartelig_15plus += tmp_artelig;
-
-	  real_type tmp_expect_mort = natural_history.cd4_mortality(hm, ha, g) *
-	    intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS);
-	  intermediate.base.expect_mort_artelig_hm(hm) += tmp_expect_mort;
-          intermediate.base.expect_mort_artelig15plus += tmp_expect_mort;
+	        real_type tmp_artelig = prop_elig * n_ba.h_hiv_adult(hm, ha, g);
+          i_ba.artelig_hahm(hm, ha - p_op.hIDX_15PLUS) = tmp_artelig;
+          i_ba.artelig_hm(hm) += tmp_artelig;
+          i_ba.Xartelig_15plus += tmp_artelig;
+
+	        real_type tmp_expect_mort = p_nh.cd4_mortality(hm, ha, g) * i_ba.artelig_hahm(hm, ha - p_op.hIDX_15PLUS);
+	        i_ba.expect_mort_artelig_hm(hm) += tmp_expect_mort;
+          i_ba.expect_mort_artelig15plus += tmp_expect_mort;
         }
 	
-        for (int hu = 0; hu < ss.hTS; ++hu) {
-          intermediate.base.Xart_15plus +=
-              state_next.base.h_art_adult(hu, hm, ha, g) +
-              dt * intermediate.base.gradART(hu, hm, ha, g);
-	}
-
+        for (int hu = 0; hu < ss_b.hTS; ++hu) {
+          i_ba.Xart_15plus += n_ba.h_art_adult(hu, hm, ha, g) +
+                              p_op.dt * i_ba.gradART(hu, hm, ha, g);
+	      }
       }
     }
 
     // calculate number on ART at end of ts, based on number or percent
-    if (pars.base.options.proj_period_int == internal::PROJPERIOD_MIDYEAR &&
-	dt * (hiv_step + 1) < 0.5) {
+    real_type art_interp_w = p_op.dt * (hiv_step + 1.0);
```

Comment:
point of interest: hopefully this is okay to define outside of any of the if statements, this was previously in the else branch